Remove zero size special case for copy_to_stream

nikic opened the pull request:
This doesn't seem to serve any purpose. Stats are expensive, so doing an unnecessary stat just to short-circuit the zero size case seems rather dubious. It can also break with stream wrappers that return inaccurate sizes (https://github.com/symfony/symfony/issues/40574) and probably can also break with stream filters. I think we should just drop it...

cmb69 reviewed (commented):
I think this is fine, but maybe add a test case for copying an empty file?

nikic commented:
Looks like we have a test for copying from an empty stream here: https://github.com/php/php-src/blob/07fa13088e1349f4b5a044faeee57f2b34f6b6e4/ext/standard/tests/streams/bug47997.phpt I didn't spot one for empty files in particular though, so I'll add one.

nicolas-grekas commented:
Should this be considered as a bugfix, thus for 7.4?

cmb69 commented:
I just found that this (https://github.com/php/php-src/commit/22517ce0de20456ced990e4ae74267a61f54d590) was introduced as bug fix, where a [comment in the bug report](https://bugs.php.net/bug.php?id=22234#1045316047) claims it would be an issue for platforms not supporting mmap(). However, when I tested on Windows, php_stream_mmap_range() failed, but the rest worked well, so maybe this has been resolved otherwise in the meantime.

nicolas-grekas commented:
Maybe not relevant, but the end of the function says:
```
	/* we've got at least 1 byte to read.
	 * less than 1 is an error */
```

nikic commented:
@nicolas-grekas Good point. This code doesn't actually get reached when the size is zero, but the comment is certainly confusing, and the special handling there is unnecessary. I've simplified that code a bit and dropped the comment.
